src/camvidlog2/bioclip/cli.py

```
@@ -0,0 +1,150 @@
+    search_embeddings = load_embedding_dataframe(
+        embedding_collection,
+        video_embeddings,
+    )
+
+    distances = calculate_distances(video_embeddings, search_embeddings, roll=roll)
+
+    distances_index = 0
+    for j, embedding_group in enumerate(embedding_collection.groups):
+        # calculate the average distance over the embeddings in this group
+        group_distances = distances.iloc[
+            :, distances_index : distances_index + len(embedding_group.items)
```

Flake8 (7.2.0)

[error] whitespace before ':'

(E203)